libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/extensions/MapboxLifecycleExtensions.kt

```diff
@@ -55,74 +52,3 @@ fun <T : MapboxNavigationObserver> LifecycleOwner.attachResumed(vararg observers
         }
     })
 }
-
-@ExperimentalPreviewMapboxNavigationAPI
-fun LifecycleOwner.attachCreated(
-    mapboxNavigation: MapboxNavigation,
-    observer: MapboxNavigationObserver
-) = attachOnLifecycle(
-    Lifecycle.Event.ON_CREATE,
-    Lifecycle.Event.ON_DESTROY,
-    mapboxNavigation,
-    observer
-)
-
-@ExperimentalPreviewMapboxNavigationAPI
-fun LifecycleOwner.attachStarted(
-    mapboxNavigation: MapboxNavigation,
-    observer: MapboxNavigationObserver
-) = attachOnLifecycle(
-    Lifecycle.Event.ON_START,
-    Lifecycle.Event.ON_STOP,
-    mapboxNavigation,
-    observer
-)
-
-@ExperimentalPreviewMapboxNavigationAPI
-fun LifecycleOwner.attachResumed(
-    mapboxNavigation: MapboxNavigation,
-    observer: MapboxNavigationObserver
-) = attachOnLifecycle(
-    Lifecycle.Event.ON_RESUME,
-    Lifecycle.Event.ON_PAUSE,
-    mapboxNavigation,
-    observer
-)
-
-@ExperimentalPreviewMapboxNavigationAPI
-fun LifecycleOwner.attachOnLifecycle(
-    attachEvent: Lifecycle.Event,
-    detachEvent: Lifecycle.Event,
-    mapboxNavigation: MapboxNavigation,
-    observer: MapboxNavigationObserver
-) {
-    lifecycle.addObserver(AttachOnLifecycle(attachEvent, detachEvent, mapboxNavigation, observer))
-}
-
-@ExperimentalPreviewMapboxNavigationAPI
-internal class AttachOnLifecycle(
-    private val attachEvent: Lifecycle.Event,
-    private val detachEvent: Lifecycle.Event,
-    private val mapboxNavigation: MapboxNavigation,
-    private val observer: MapboxNavigationObserver
-) : LifecycleEventObserver {
-
-    private var attached = false
-
-    override fun onStateChanged(source: LifecycleOwner, event: Lifecycle.Event) {
-        if (event == attachEvent) attach()
-        if (event == detachEvent || event == Lifecycle.Event.ON_DESTROY) detach()
-    }
-
-    private fun attach() {
-        if (attached) return
-        attached = true
-        observer.onAttached(mapboxNavigation)
-    }
-
-    private fun detach() {
-        if (!attached) return
-        attached = false
-        observer.onDetached(mapboxNavigation)
-    }
-}
```

Comment:
Please revert all these changes. The above extensions and classes are a critical part of ComponentInstaller API.

Reply:
I still think we should remove these extensions.

Anything installed onto `MapboxNavigation` directly, will have issues integrating with `:androidauto`. There are also conflicts with existing `LifecycleOwner` extensions